Documentation/FileFormats.md

```diff
@@ -0,0 +1,58 @@
+# Serenity Formats
+
+This document summarizes all non-standard file formats, data formats, and mime-types used by Serenity.
+
+## glyph/x-fonteditor (Clipboard-only)
+
+Requires the metadata-fields, `char` (usually encoded as a string of length 1: the codepoint as a character),
+`width`, and `height` (encoded as a decimal number).
+
+The data is encoded in width times height many bytes, either 0 (clear) or 1 (set).
+
+## image/x-serenityos (Clipboard-only)
+
+Requires the metadata-fields `width`, `height`, `scale`, `format` (see `Gfx::BitmapFormat`), and `pitch`, encoded as decimal strings.
+
+The data is encoded according to `Gfx::determine_storage_format(BitmapFormat)`, so either as
+BGRx8888, BGRA8888, RGBA8888, or 8-bit pallette index. Note that the pallette is not transferred.
+
+## text/uri-list (Clipboard-only)
+
+Newline-delimited set of URIs.
+
+## Application File (`*.af` files)
+
+These files are human-readable and are a subset of the INI-format, have no easily detectable filemagic, and define System Menu entries.
+
+## Font (`*.font` files)
+
+These files contain bitmap definitions of fonts, either varying-width or fixed-width.
+The header definition can be found in `Gfx::FontFileHeader` in `Userland/Libraries/LibGfx/BitmapFont.cpp`.
+Most prominently, the first four bytes contain the filemagic: `!Fnt`.
+
+## GUI Markup Language (`*.gml` files)
+
+These files are human-readable, have no easily detectable filemagic, and define GUI interfaces.
+The format is strongly influenced by QML, the Qt Modeling Lange.
```

Comment:
```suggestion
The format is strongly influenced by QML, the Qt Modeling Language.
```

Reply:
It's typos like these that make me quest my sanity ;)